satellite/satellitedb/overlaycache.go

```diff
@@ -618,12 +618,46 @@ func (cache *overlaycache) UpdateBatch(ctx context.Context, updateReqList []*ove
 // CreateEntryIfNotExists creates a overlay node entry and saves to overlay if it didn't already exist
 func (cache *overlaycache) CreateEntryIfNotExists(ctx context.Context, nodeID storj.NodeID) (stats *overlay.NodeStats, err error) {
 	defer mon.Task()(&ctx)(&err)
+	tx, err := cache.db.Open(ctx)
+	if err != nil {
+		return nil, Error.Wrap(err)
+	}
 
-	getStats, err := cache.GetStats(ctx, nodeID)
-	if err == ErrNodeNotFound {
-		return cache.Create(ctx, nodeID, nil)
+	dbNode, err := tx.Find_Node_By_Id(ctx, dbx.Node_Id(nodeID.Bytes()))
+	if err != nil {
+		return nil, Error.Wrap(utils.CombineErrors(err, tx.Rollback()))
 	}
-	return getStats, err
+
+	if dbNode == nil {
+		var (
+			totalAuditCount    int64
+			auditSuccessCount  int64
+			auditSuccessRatio  float64
+			totalUptimeCount   int64
+			uptimeSuccessCount int64
+			uptimeRatio        float64
+			wallet             string
+			email              string
+		)
+		dbNode, err = tx.Create_Node(
```

Comment:
Do we normally create nodes with no other information such as wallet and email? Is it possible that these nodes could get used without getting paid?

Reply:
the wallet and email are updated separately. 

discovery/service.go L#157
// update wallet with correct info
		info, err := discovery.kad.FetchInfo(ctx, *node)